lib/model/query/submissions.js

```diff
@@ -65,13 +65,14 @@ const createVersion = (partial, deprecated, form, deviceIdIn = null, userAgentIn
 
   // we already do transactions but it just feels nice to have the cte do it all at once.
   return one(sql`
-with logical as (update submissions set "reviewState"='edited', "updatedAt"=clock_timestamp()
-  where id=${deprecated.submissionId})
+with logical as (update submissions set "reviewState"='edited', "updatedAt"=clock_timestamp() 
+  where id=${deprecated.submissionId} returning "reviewState","updatedAt")
 , upd as (update submission_defs set current=false where "submissionId"=${deprecated.submissionId})
-${_defInsert(deprecated.submissionId, partial, form.def.id, actorId, null, deviceId, userAgent)}`)
+, def as (${_defInsert(deprecated.submissionId, partial, form.def.id, actorId, null, deviceId, userAgent)})
+select def.*, logical.* from def, logical`)
     // TODO/HACK: lame that we are reconstructing this this way.
-    .then((saved) => new Submission({ instanceId: partial.instanceId },
-      { def: new Submission.Def(saved), xml: new Submission.Xml({ xml: partial.xml }) }));
+    .then(({reviewState, updatedAt, ...savedDef}) =>  new Submission({ instanceId: partial.instanceId, reviewState, updatedAt },
```

Comment:
Overall, I think this was the right approach for what we thought the problem was -- getting the right info from the database and passing it along to the hacky `Submission` constructor.

But maybe it shouldn't be constructing a `Submission` here, just a `Submission.Def`? Because that is more consistent with what we expect to be returned? 

... Hmm, briefly tried that myself and it makes the step that follows https://github.com/getodk/central-backend/blob/master/lib/resources/submissions.js#L204 fail because it is expecting to work with a `Submission`. But maybe line 205 right after could return `submission.def` instead? And/or we could just add more explanatory comments about what we've learned along this journey...

Reply:
Just to follow up on this comment, L204 fails because `SubmissionAttachments.create()` expect XML, not just a `Submission.Def`. However, like @ktuite said, if we change L205 to `submission.def` instead of `submission`, I think that would make the intention of that endpoint clearer. If there are places where additional comments would make things clearer, those would be great as well.